firefox-ios/Client/Frontend/PasswordGenerator/View/PasswordGeneratorPasswordFieldView.swift

```diff
@@ -0,0 +1,103 @@
+        button.accessibilityLabel = .PasswordGenerator.RefreshPasswordButtonA11yLabel
+        button.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.passwordRefreshButton
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.layer.borderWidth = UX.passwordFieldBorderWidth
+        self.layer.cornerRadius = UX.passwordFieldCornerRadius
+        self.accessibilityIdentifier = AccessibilityIdentifiers.PasswordGenerator.passwordField
+        setupLayout()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupLayout() {
+        self.addSubviews(passwordLabel, passwordRefreshButton)
+        NSLayoutConstraint.activate([
+            passwordLabel.leadingAnchor.constraint(
+                equalTo: self.leadingAnchor,
+                constant: UX.passwordFieldHorizontalPadding),
+            passwordLabel.topAnchor.constraint(
+                equalTo: self.topAnchor,
+                constant: UX.passwordFieldVerticalPadding),
+            passwordLabel.bottomAnchor.constraint(
+                equalTo: self.bottomAnchor,
+                constant: -UX.passwordFieldVerticalPadding),
+            passwordRefreshButton.leadingAnchor.constraint(
+                greaterThanOrEqualTo: passwordLabel.trailingAnchor,
+                constant: UX.passwordLabelAndButtonSpacing),
+            passwordRefreshButton.trailingAnchor.constraint(
+                equalTo: self.trailingAnchor,
+                constant: -UX.passwordFieldHorizontalPadding),
+            passwordRefreshButton.centerYAnchor.constraint(
+                equalTo: self.centerYAnchor),
+            passwordRefreshButtonWidthConstraint,
+            passwordRefreshButtonHeightConstraint,
+        ])
+    }
+
+    func configure(password: String) {
+        passwordLabel.text = password
+        passwordLabel.accessibilityAttributedLabel = generateAccessibilityAttributedLabel(password: password)
+    }
+
+    private func generateAccessibilityAttributedLabel(password: String) -> NSMutableAttributedString {
+        let fullString = String(format: .PasswordGenerator.PasswordReadoutPrefaceA11y, password)
+        let attributedString = NSMutableAttributedString(string: fullString)
+        let rangeOfPassword = (attributedString.string as NSString).range(of: password)
+        attributedString.addAttributes([.accessibilitySpeechSpellOut: true], range: rangeOfPassword)
+        return attributedString
+    }
+
+    func applyTheme(theme: any Common.Theme) {
+        self.backgroundColor = theme.colors.layer2
+        self.layer.borderColor = theme.colors.borderPrimary.cgColor
+        passwordLabel.textColor = theme.colors.textPrimary
+        passwordRefreshButton.tintColor = theme.colors.iconPrimary
+    }
+
+    func applyDynamicFontChange() {
+        passwordLabel.font = FXFontStyles.Regular.body.scaledFont()
```

Comment:
Nit: The font doesn't have to be set again when the dynamic font size changes - it adjusts automatically.